### DISCUSSION TITLE
Querying a the database using rpc to call a function returns empty data. Can anyone help out?

### DISCUSSION THREAD

**Discussion by rxmii4269** [supabase/supabase#5136] [Questions] [answered]
I created a function on postgresql and when querying this function using rpc() an empty array is returned even when data is there. Calling the function in the SQL editor does indeed return data, so it is confirmed that the function works as it should.
Any help is much appreciated.

`CREATE OR REPLACE FUNCTION public.tweets_of_who_i_follow(uid uuid) RETURNS TABLE(id bigint, user_id uuid, created_at timestamp with time zone, description text, has_photo boolean, photo_name character varying, visibility character varying) AS $$
 BEGIN
	RETURN QUERY
	SELECT tweets.*
 	FROM tweets  
 	INNER JOIN followers ON tweets.user_id = followers.user_id
 	WHERE followers.user_id = uid;
	END;	
$$ LANGUAGE plpgsql;`
![image](https://user-images.githubusercontent.com/49266906/150813567-0264272c-d432-474e-81e1-5324a6e921af.png)

**Comment by GaryAustin1** [comment, marked as answer]
At a minimum verify your RLS (if on) works for your user.  The SQL editor superuser bypasses RLS.
OR you can add "security definer" to your function if you want it to bypass RLS.

Edit: also you tried to hide your site URL but it is at top of your photo...

**Reply by rxmii4269** [reply]
l see I did not create any policies for these newly created tables I have added, my bad. Upon enable read access to all users I am now receiving data. Thanks for pointing me in the right place.

PS: I noticed that my attempt to hide the url is pointless so I didn't bother to do that one lol
